crates/sargon/src/types/hd_signature.rs

```diff
@@ -3,9 +3,9 @@ use crate::prelude::*;
 /// A signature of `intent_hash` by `entity` using `factor_source_id` and `derivation_path`, with `public_key` used for verification.
 #[derive(Clone, PartialEq, Eq, Hash, derive_more::Debug)]
 #[debug("HDSignature {{ input: {:#?} }}", input)]
-pub struct HDSignature {
+pub struct HDSignature<S: Signable> {
```

Comment:
Looks like this can be made generic over `T: SignableID` only, not the whole `Signable` trait? Given my review comment of `HDSignatureInput` ?